taskcluster/async/asyncclient.py

```diff
@@ -0,0 +1,100 @@
+"""This module is used to interact with taskcluster rest apis"""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+import asyncio
+import aiohttp
+
+from contextlib import contextmanager
+
+import taskcluster.exceptions as exceptions
+import taskcluster.baseclient as baseclient
+import taskcluster.utils as utils
+import taskcluster.asyncutils as asyncutils
+
+log = logging.getLogger(__name__)
+
+
+class AsyncClient(baseclient.BaseClient):
+    """Async class for API Client Classes. Each individual Client class
+    needs to set up its own methods for REST endpoints and Topic Exchange
+    routing key patterns.
+    """
+
+    @contextmanager
+    def contextSession(self, *args, **kwargs):
+        yield self.createSession(*args, **kwargs)
+
+    def createSession(self, session=None, args=(), kwargs=None):
```

Comment:
When using pylint (we don't here), it throws an error whenever an external object calls a single underscore prefixed method, because those are "private".  This causes enough headaches for tests that I default to open unless there's a reason for making something private.

Since we don't use pylint here, I don't have a strong opinion, but still default to open.